satellite/metabase/rangedloop/service.go

```diff
@@ -195,30 +201,79 @@ func finishObservers(ctx context.Context, observerStates []observerState) (obser
 
 // Iterating over the segments is done.
 // This is the reduce step.
-func finishObserver(ctx context.Context, state observerState) (ObserverDuration, error) {
+func finishObserver(ctx context.Context, log *zap.Logger, state observerState) ObserverDuration {
+	if state.err != nil {
+		return ObserverDuration{
+			Observer: state.observer,
+			Duration: -1 * time.Second,
```

Comment:
won't 0 have the same effect here (and in other places)? that observer have no duration?

Reply:
I think -1 seems more explicitly an error than 0, which could be a valid duration.